src/test/java/org/takes/rs/RsPrintTest.java

```diff
@@ -43,4 +51,54 @@ public void failsOnInvalidHeader() throws IOException {
         new RsPrint(new RsWithHeader("name", "\n\n\n")).print();
     }
 
+    /**
+     * RsPrint can flush body contents even when exception happens.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void flushBodyEvenWhenExceptionHappens() throws IOException {
+        final OutputStream output = Mockito.mock(OutputStream.class);
+        final IOException exception = new IOException("Error");
+        Mockito.doThrow(exception).when(output)
+            .write(
+                Mockito.any(byte[].class),
+                Mockito.anyInt(),
+                Mockito.anyInt()
+        );
+        try {
+            new RsPrint(new RsText("Hello"))
+                .printBody(output);
+        } catch (final IOException ex) {
+            if (!ex.equals(exception)) {
+                throw ex;
+            }
+        }
+        Mockito.verify(output).flush();
+    }
+
+    /**
+     * RsPrint can flush head contents even when exception happens.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    @Ignore
+    public void flushHeadEvenWhenExceptionHappens() throws IOException {
+        final OutputStream output = Mockito.mock(OutputStream.class);
+        final IOException exception = new IOException("Failure");
+        Mockito.doThrow(exception).when(output)
```

Comment:
@t-izbassar same about "Mockito" here. Further, you can reuse fake class from first test.